pkg/chains/storage/oci/oci_test.go

```diff
@@ -14,23 +14,118 @@ limitations under the License.
+
 	"github.com/google/go-containerregistry/pkg/authn"
+	"github.com/google/go-containerregistry/pkg/authn/k8schain"
+	"github.com/google/go-containerregistry/pkg/registry"
 	"github.com/google/go-containerregistry/pkg/v1/remote"
 	"github.com/in-toto/in-toto-golang/in_toto"
-	"github.com/tektoncd/chains/pkg/config"
+	slsa "github.com/in-toto/in-toto-golang/in_toto/slsa_provenance/v0.2"
+	"github.com/sigstore/sigstore/pkg/signature/payload"
 	"github.com/tektoncd/pipeline/pkg/apis/pipeline/v1beta1"
-	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	remotetest "github.com/tektoncd/pipeline/test"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes/fake"
 	logtesting "knative.dev/pkg/logging/testing"
 )
 
+const (
+	namespace      = "oci-test"
+	serviceAccount = "oci-test-sa"
+)
+
+var (
+	sa = &corev1.ServiceAccount{
+		ObjectMeta: metav1.ObjectMeta{
+			Namespace: namespace,
+			Name:      serviceAccount,
+		},
+	}
+
+	tr = &v1beta1.TaskRun{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "foo",
+			Namespace: namespace,
+		},
+	}
+)
+
 func TestBackend_StorePayload(t *testing.T) {
+	ctx := context.Background()
+	kc, err := k8schain.New(ctx, fake.NewSimpleClientset(sa), k8schain.Options{
+		Namespace:          namespace,
+		ServiceAccountName: serviceAccount,
+	})
+	if err != nil {
+		t.Fatal(err)
+	}
+	auth := remote.WithAuthFromKeychain(authn.DefaultKeychain)
+
+	// Create registry server
+	s := httptest.NewServer(registry.New())
+	defer s.Close()
+	u, _ := url.Parse(s.URL)
+
+	// Push image to server
+	ref, err := remotetest.CreateImage(u.Host+"/task/"+tr.Name, tr)
+	if err != nil {
+		t.Fatalf("failed to push img: %v", err)
+	}
+	digest := strings.Split(ref, "@")[1]
+	digSplits := strings.Split(digest, ":")
+	algo, hex := digSplits[0], digSplits[1]
 
-	// pretty much anything that has no Subject
-	sampleIntotoStatementBytes, _ := json.Marshal(in_toto.Statement{})
-	logger := logtesting.TestLogger(t)
+	simple := simple.SimpleContainerImage{
+		Critical: payload.Critical{
+			Identity: payload.Identity{
+				DockerReference: u.Host + "/task/" + tr.Name,
+			},
+			Image: payload.Image{
+				DockerManifestDigest: digest,
+			},
+			Type: payload.CosignSignatureType,
+		},
+	}
+	simplebytes, _ := json.Marshal(simple)
+	if err != nil {
+		t.Fatalf("failed to marshal: %v", err)
+	}
+
+	intotoStatement := in_toto.ProvenanceStatement{
+		StatementHeader: in_toto.StatementHeader{
+			Type:          in_toto.StatementInTotoV01,
+			PredicateType: slsa.PredicateSLSAProvenance,
+			Subject: []in_toto.Subject{
+				{
+					Name: u.Host + "/task/" + tr.Name,
+					Digest: slsa.DigestSet{
+						algo: hex,
+					},
+				},
+			},
+		},
+		Predicate: slsa.ProvenancePredicate{},
+	}
+	intotoStatementBytes, err := json.Marshal(intotoStatement)
+	if err != nil {
+		t.Fatalf("failed to marshal: %v", err)
+	}
+
+	sampleIntotoStatementBytes, err := json.Marshal(in_toto.Statement{})
```

Comment:
nit: Since you're doing the json marshal for each message type, you could change `rawPayload []byte` -> `payload interface{}` and marshal it below!

Reply:
yes! thanks!